examples/docker/hf_model.py

```diff
@@ -0,0 +1,42 @@
+from pipeline import docker
+from pipeline import Pipeline, Variable, pipeline_model, pipeline_function
+from transformers import GPTNeoForCausalLM, GPT2Tokenizer
+
+
+@pipeline_model
+class PipelineGPTNeo:
+    def __init__(self):
+        self.model = None
+        self.tokenizer = None
+
+    @pipeline_function
+    def predict(self, input_data: str) -> str:
+        if self.model == None:
+            self.model = GPTNeoForCausalLM.from_pretrained("EleutherAI/gpt-neo-125M")
+            self.tokenizer = GPT2Tokenizer.from_pretrained("EleutherAI/gpt-neo-125M")
+
+        input_ids = self.tokenizer(input_data, return_tensors="pt").input_ids
+
+        gen_tokens = self.model.generate(
+            input_ids,
+            do_sample=True,
+            temperature=0.9,
+            max_length=100,
+        )
+        gen_text = self.tokenizer.batch_decode(gen_tokens)[0]
+        return gen_text
+
+
+with Pipeline("GPTNeo") as builder:
+    in_1 = Variable(str, is_input=True)
+    builder.add_variables(in_1)
+
+    gpt_neo = PipelineGPTNeo()
+
+    out_str = gpt_neo.predict(in_1)
+
+    builder.output(out_str)
+
+gpt_neo_pipeline = Pipeline.get_pipeline("GPTNeo")
+print(gpt_neo_pipeline.run("lol how are you mi amigo"))
```

Comment:
you can do `Pipeline.run_local("GPTNeo", *args)` as well

Reply:
Is there a need to have to 'fetch' the pipeline object? Could one also do `builder.run(…)`?

Reply:
from builder you can also do `builder.run_local(id, *args)`

Reply:
Seems nice. Is there a reason to re-fetch when that's possible then?

Reply:
It's just a preference, and specifically here because I pass the pipeline as an arg to the docker building function the line after.
Personally I don't like the `.run_local` as it implies that there should be a `.run_remote` or an equivalent. The base pipeline features should always be locally focused.

Reply:
yeah, I added a `Pipeline.run` which is remote, I think I misunderstood this as we wanted users to run on our cloud instead of locally by default

Reply:
also if Pipeline managed the calls, you wouldn't need the extra fetch:
```
Pipeline.run(id: str, *args)
Pipeline.create_api(ids:List[str])
```

Reply:
otherwise, for creating an api with many Graphs, you would need multiple fetches, would become quite verborragic, I think